### PR TITLE
feat(agent): support batched array loops

### LINKED ISSUE #4664: [Feature] 目前ReactAgent搭配Sequential Agent这些智能编排不支持循环调用
### Feature Request

比如我在编排智能体的时候，其中有一个智能体是分析图片的，我想让他一次只分析5张图片，但是传参可能20张，就需要分批处理，
目前解决方案
1. 放弃智能编排自己写
2.通过ModelInterceptor（不合理）
3.通过tools  （不可控）
4. 写两个Agent，一个Agent正常处理5张图，另外的Agent 重写call方法，循环调用第一个Agent
做分批处理的逻辑和结果整合

目前用的方案四，感觉编排封装这一块不太灵活

### Is your feature request related to a problem? Please describe

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by March-77**
### Describe what this PR does / why we need it

Adds opt-in batched dispatch to LoopAgent array loops. Callers can now use `LoopMode.array(batchSize)` when a sub-agent should process several input elements per iteration, such as processing 20 images in groups of 5.

The root cause of the limitation was that `ArrayLoopStrategy.loopDispatch` always selected exactly one list element and incremented its cursor by one. There was no API for controlling the number of elements sent to each iteration.

### Does this pull request fix one issue?

Fixes #4664

### Describe how you did it

- Added `LoopMode.array(int batchSize)` and `LoopMode.array(int batchSize, Converter<...>)` overloads.
- Added validated batch-size configuration to `ArrayLoopStrategy`.
- Dispatches each opt-in batch as a valid JSON array and advances the cursor by the actual number of consumed elements.
- Preserves the existing `array()` and `array(converter)` single-element message behavior.
- Added deterministic unit tests for full batches, a short final batch, backward-compatible single-element dispatch, completion, and invalid batch sizes.

User impact: LoopAgent users can express bounded batch processing without wrapping or subclassing agents. Existing users see no behavior change unless they select a batched overload.

### Describe how to verify it

```bash
JAVA_HOME=/opt/homebrew/opt/openjdk PATH=/opt/homebrew/opt/openjdk/bin:$PATH ./mvnw -pl spring-ai-alibaba-agent-framework test
```

Result: BUILD SUCCESS; 593 tests run, 0 failures, 0 errors, 167 environment-dependent tests skipped. Checkstyle and Spotless passed in the same build.

### Special notes for reviews

The API change is additive and source compatible. Batched message serialization is only enabled when `batchSize` is explicitly provided; the final batch may contain fewer elements. Non-positive sizes fail fast with `IllegalArgumentException`.

### CODE REVIEW

File: spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/flow/agent/loop/ArrayLoopStrategy.java [hunk 1]
```diff
@@ -63,8 +77,12 @@ public Map<String, Object> loopDispatch(OverAllState state) {
         List<?> list = state.value(loopListKey(), List.class).orElse(List.of());
         int index = state.value(loopCountKey(), maxLoopCount());
         if(index < list.size()) {
-            UserMessage message = new UserMessage(list.get(index).toString());
-            return Map.of(loopCountKey(), index + 1, loopFlagKey(), true,
+            int nextIndex = Math.min(index + batchSize, list.size());
+            String messageText = batchSize == 1
+                    ? list.get(index).toString()
+                    : JsonParser.toJson(list.subList(index, nextIndex));
```
[the remark this answers is not in the document]
Comment (March-77) [reply spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/flow/agent/loop/ArrayLoopStrategy.java]: Fixed in `e3345d107`. Explicit batch constructors now retain JSON-array framing even when the configured size is one, while the legacy no-size factories keep scalar messages. The regression covers both `LoopMode.array(1)` and the size-one converter overload; all 4 strategy tests pass and Checkstyle is clean.

**Review by oss-sentinel-ai — COMMENTED** [review]
### Summary
Reviewed the changes in this PR (191 lines changed). The changes appear to be well-structured.

### Observations
- Changes span 191 lines
- Please ensure all edge cases are covered
- Verify test coverage for the modified code paths

---
*Automated review by github-manager-bot*

**Comment by March-77** [comment]
@yuluo-yx 已重新核对当前 head、最新 main 变化、mergeability、CI、reviews 和 review threads；现有评审意见均已验证并清理，未发现需要追加代码修改的问题。烦请在方便时进行维护者审查，谢谢。
